src/modules/accounting/prediction.py

```diff
@@ -0,0 +1,61 @@
+from typing import List, Iterable
+
+import web3.eth
+from web3 import Web3
+from web3.contract import ContractEvent
+from web3.types import Wei, EventData
+
+from modules.accounting.constants import REWARDS_HISTORICAL_PERIOD_IN_SECONDS, SECONDS_PER_SLOT
+from src.typings import BlockStamp
+
+
+class Prediction:
+    _eth_distributed: ContractEvent
+    _eth_distributed_address: web3.eth.Address = 'address'
+    _eth_distributed_abi = 'abi'
+
+    def __init__(self, w3: Web3):
+        self._eth_distributed = w3.eth.contract(
+            address=self._eth_distributed_address,
+            abi=self._eth_distributed_abi
+        ).events.ETHDistributed
+
+    def get_rewards_per_epoch(self, blockstamp: BlockStamp) -> Wei:
+        from_block, to_block = self.get_block_for_filtering(blockstamp['block_number'])
+
+        logs: Iterable[EventData]
+
+        logs = self._eth_distributed.build_filter().get_logs(
+            from_block=from_block,
+            to_block=to_block,
+        )
+
+        rewards: List[Wei] = list()
+        for log in logs:
+            reward = log['args']['withdrawalsWithdrawn'] + log['args']['executionLayerRewardsWithdrawn']
+            rewards.append(Wei(reward))
+
+        median: Wei = self.median(rewards)
+        return Wei(median)
+
+    @staticmethod
+    def get_block_for_filtering(ref_block: int) -> (int, int):
+        # Rough estimate, since there could be missing blocks in the slots
+        # 604800 / 12 = 50400
+        max_period_block_range = REWARDS_HISTORICAL_PERIOD_IN_SECONDS / SECONDS_PER_SLOT
```

Comment:
Why this constant in seconds, not in Slots?

Reply:
Or epochs

Reply:
Discussed here https://discord.com/channels/761182643269795850/1072446086423257088/1073220253473050767